frontend/pages/index.js

```diff
@@ -12,11 +70,38 @@ const Home = () => {
         />
         <link rel="icon" href="/favicon.ico" />
       </Head>
-      <Section>
-        {/*
-          PAGE CONTENT GOES HERE
-        */}
-      </Section>
+      {showAddBookModal && (
+        <AddBookModal
+          onClose={onAddBookModalToggle}
+          onSubmit={onAddBookModalSubmit}
+        />
+      )}
+      <Layout>
+        <Section>
+          <Flex
+            justifyContent="space-between"
+            alignItems="center"
+            p={`0 0 ${spacing["2.5"]} 0`}
+            width="100%"
+          >
+            <Text as="h3">My Library</Text>
+            <Button onClick={onAddBookModalToggle}>
+              <Text fontWeight={fontWeight.bold}>+ Add Book</Text>
+            </Button>
+          </Flex>
+          <Grid
+            width="100%"
+            gap={spacing["0.75"]}
+            gridTemplateColumns="repeat(auto-fill, 276px)"
+          >
+            {booksLoading || booksError || !books?.length ? (
+              <EmptyState loading={booksLoading} error={booksError} />
+            ) : (
```

Comment:
Please elaborate on the approach used here... remember, if you can keep JSX lean - we want to keep it lean.